homeassistant/components/airq/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""The air-Q integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from aioairq import AirQ
+from aiohttp.client_exceptions import ClientError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, MANUFACTURER, TARGET_ROUTE, UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+class AirQCoordinator(DataUpdateCoordinator):
+    """Coordinator is responsible for querying the device at a specified route."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialise a custom coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=UPDATE_INTERVAL),
+        )
+        session = async_get_clientsession(hass)
+        self.airq = AirQ(
+            entry.data[CONF_IP_ADDRESS], entry.data[CONF_PASSWORD], session
+        )
+        self.device_id = entry.unique_id
+        assert self.device_id is not None
+        self.device_info = DeviceInfo(
+            manufacturer=MANUFACTURER,
+            identifiers={(DOMAIN, self.device_id)},
+        )
+        self.device_info.update(entry.data["device_info"])
+
+    async def _async_update_data(self) -> dict:
+        """Fetch the data from the device."""
+        try:
+            data = await self.airq.get(TARGET_ROUTE)
+        except ClientError as err:
+            raise UpdateFailed(f"Error while retrieving data: {err}") from err
+        return self.airq.drop_uncertainties_from_data(data)
```

Comment:
Catching `ClientError` is redundant here. As I said earlier, some common exceptions, including `aiohttp` errors, will be [caught by the coordinator automatically](https://github.com/home-assistant/core/blob/f3e432c9c7735daddd2946773cc6572567821a5e/homeassistant/helpers/update_coordinator.py#L204-L219).
```suggestion
        data = await self.airq.get(TARGET_ROUTE)
        return self.airq.drop_uncertainties_from_data(data)
```

Reply:
Right, thanks!